apps/website/src/server/db/short-links.ts

```diff
@@ -0,0 +1,54 @@
+import { z } from "zod";
+import { TRPCError } from "@trpc/server";
+import { prisma } from "@/server/db/client";
+import { convertToSlug, SLUG_REGEX } from "@/utils/slugs";
+
+export const shortLinkSchema = z.object({
+  label: z.string().min(1),
+  slug: z.string().regex(SLUG_REGEX).optional(),
+  link: z.string().url(),
+});
+
+export type ShortLinkSchema = z.infer<typeof shortLinkSchema>;
+
+export const existingShortLinkSchema = shortLinkSchema.and(
+  z.object({
+    id: z.string().cuid(),
+  }),
+);
+
+export async function createShortLink(input: z.infer<typeof shortLinkSchema>) {
+  const slug = convertToSlug(input.slug || input.label);
```

Comment:
I'm tempted to say this logic should live in the client, so the logic here is just worrying about writing to the DB, and the client handles this fallback?